test/order_product.test.js

```diff
@@ -0,0 +1,145 @@
+
+const { assert: { equal, deepEqual, isFunction, isObject, isArray, isNumber } } = require("chai");
+const { getAllOrderProducts, postOneOrderProduct, getOneOrderProduct, getLastOrderProduct } = require("../app/models/Order_Product");
+const { addProductToOrder } = require("../app/controllers/order_productCtrl");
+const createOrderProductTable = require("../db/order_product_table");
+
+
+// Order_Product Model
+// get All Order Products
+describe("get order products", () => {
+    describe("get all order products", () => {
+        it("should be a function", () => {
+            isFunction(getAllOrderProducts);
+        });
+
+        it("should return an array", () => {
+            getAllOrderProducts().then(data => {
+                isArray(data);
+            })
+            .catch((err) => {
+                console.log('Get All Order Products Should Return an Array', err);
+            });
+        });
+
+        it("should be an object in the 0 index of the array", () => {
+            getAllOrderProducts().then(data => {
+                isObject(data[0]);
+            })
+            .catch((err) => {
+                console.log('Get All Order Products Should be an object in the 0 index of the array', err);
+            });
+        });
+    });
+    describe("get one order product", () => {
+        it("should be a function", () => {
+            isFunction(getOneOrderProduct)
+        });
+        it("should return an object", () => {
+            getOneOrderProduct(1).then(op => {
+                isObject(op);
+            })
+            .catch((err) => {
+                console.log('Get One Order Product Should Return an object', err);
+            });
+        });
+        it("should return an object with the correct Line ID", () => {
+            getOneOrderProduct(1).then(op => {
+                equal(1, op.line_id)
+            })
+            .catch((err) => {
+                console.log('Get One Order Product Should Return an object with the correct Line Id', err);
+            });
+        });
+    });
+    describe("Get last Line ID on Order Product Table", () => {
+        it("should be a function", () => {
+            isFunction(getLastOrderProduct)
+        });
+        it("should return an integer", () => {
+            getLastOrderProduct().then(op => {
+                isNumber(op);
+            })
+            .catch((err) => {
+                console.log('Get last Line ID Should Return an Integer', err);
+            });
+        });
+    });
+});
+
+//POST New Order Product
+describe("post one order proudct", () => {
+    after(done => {
+        createOrderProductTable().then(() => {
+            done();
+        });
+    });
+    let newOP = {
+        quantity: 10,
+        order_id: 5,
+        product_id: 6,
+        price: 5
```

Comment:
I think you might need to take out `price` here, and change quantity to `order_quantity`.